cmake/pedantic-and-modern.cmake

```diff
@@ -1,24 +1,18 @@
-set(PEDANTIC_FLAGS)
-set(PEDANTIC_DEFINITIONS)
+# link to this library if you want your code to be compiled in a strict/pedantic mode
```

Comment:
This is a great way to achieve this! I do like the set-and-forget of "just setting all the flags" but this is a great compromise.